pkg/user/appointee/set.go

```diff
@@ -59,29 +63,97 @@ func setAppointeePermission(
 	if err != nil {
 		return err
 	}
-	receipt, err := permissionWriter.SetPermission(
-		ctx,
-		elcontracts.SetPermissionRequest{
-			AccountAddress:   config.AccountAddress,
-			AppointeeAddress: config.AppointeeAddress,
-			Target:           config.Target,
-			Selector:         config.Selector,
-			WaitForReceipt:   true,
-		},
-	)
+	return broadcastOrPrint(ctx, logger, permissionWriter, config)
+}
+
+func broadcastOrPrint(
+	ctx context.Context,
+	logger logging.Logger,
+	permissionWriter SetAppointeePermissionWriter,
+	config *setConfig) error {
+	var err error
+	permissionRequest := elcontracts.SetPermissionRequest{
+		AccountAddress:   config.AccountAddress,
+		AppointeeAddress: config.AppointeeAddress,
+		Target:           config.Target,
+		Selector:         config.Selector,
+		WaitForReceipt:   true,
+	}
+	if config.Broadcast {
+		err = broadcastSetAppointeeCallData(ctx, permissionWriter, config, permissionRequest)
+	} else {
+		err = printSetAppointeeCallData(logger, permissionWriter, config, permissionRequest)
+	}
+	return err
+}
+
+func broadcastSetAppointeeCallData(
+	ctx context.Context,
+	permissionWriter SetAppointeePermissionWriter,
+	config *setConfig,
+	request elcontracts.SetPermissionRequest,
+) error {
+	receipt, err := permissionWriter.SetPermission(ctx, request)
+	if err == nil {
+		common.PrintTransactionInfo(receipt.TxHash.String(), config.ChainID)
+	}
+	return err
```

Comment:
```suggestion
	if err != nil {
		return err
	}
	common.PrintTransactionInfo(receipt.TxHash.String(), config.ChainID)
	return nil
```

This is more consistent pattern of `err != nil` otherwise it might create confusion later